augur/io/sqlite3.py

```diff
@@ -0,0 +1,137 @@
+import os
+import pandas as pd
+import sqlite3
+from typing import List
+
+
+class Sqlite3Database:
+    def __init__(self, file:str=''):
+
+        self.file = file
+        """Database file. Empty string if using an in-memory database."""
+
+        self.use_in_memory_db = not self.file
+        """Whether to use an in-memory database."""
+
+        if self.use_in_memory_db:
+            # More info: https://www.sqlite.org/inmemorydb.html
+            self.in_memory_db_connection = sqlite3.connect(":memory:")
+            """Persistent connection to an in-memory database."""
```

Comment:
This is one of a few features that aren't used in the current PR, but are used in #854. I'd be happy to remove them from here if they seem unnecessary.

Reply:
I'm fine to keep this here prospectively.

Reply:
I'm planning to split all prospective features into separate commits, for example this one is now a7f322aabf0bdcd1d608282c0cd8cb2606fd53de.